athenahealth/patients.go

```diff
@@ -444,3 +446,82 @@ func (h *HTTPClient) UpdatePatientInformationVerificationDetails(ctx context.Con
 
 	return nil
 }
+
+// GetPatientCustomFields - Retrieves custom fields information for a specific patient.
+// GET /v1/{practiceid}/patients/{patientid}/customfields
+// https://docs.athenahealth.com/api/api-ref/patient-custom-fields#Get-custom-field-information-from-patient's-records
+func (h *HTTPClient) GetPatientCustomFields(ctx context.Context, patientID, departmentID string) ([]*CustomFieldValue, error) {
+	out := []*CustomFieldValue{}
+
+	query := url.Values{}
+	query.Add("departmentid", departmentID)
+
+	_, err := h.Get(ctx, fmt.Sprintf("/patients/%s/customfields", patientID), query, &out)
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+type updatePatientCustomFieldsResponse struct {
+	Success         bool `json:"success"`
+	UpdatedCount    int  `json:"updatedCount"`
+	DisallowedCount int  `json:"disallowedCount"`
+}
+
+// UpdatePatientCustomFields - Update custom-field information from patient's records.
+// PUT /v1/{practiceid}/patients/{patientid}/customfields
+// https://docs.athenahealth.com/api/api-ref/patient-custom-fields#Update-custom-field-information-from-patient's-records
+func (h *HTTPClient) UpdatePatientCustomFields(ctx context.Context, patientID, departmentID string, customFields []*CustomFieldValue) error {
+	out := &updatePatientCustomFieldsResponse{}
+
+	customFieldsJSON, err := json.Marshal(customFields)
+	if err != nil {
+		return errors.Wrap(err, "marshaling custom fields")
+	}
+
+	form := url.Values{}
+	form.Add("departmentid", departmentID)
+	form.Add("customfields", string(customFieldsJSON))
+
+	_, err = h.PutForm(ctx, fmt.Sprintf("/patients/%s/customfields", patientID), form, out)
+	if err != nil {
+		return err
+	}
+
+	if !out.Success {
+		return errors.New("unexpected response")
+	}
+
+	return nil
+}
+
+type ListPatientsMatchingCustomFieldResult struct {
+	Patients []*Patient
+
+	Pagination *PaginationResult
+}
+
+type listPatientsMatchingCustomFieldResponse struct {
+	Patients []*Patient `json:"patients"`
+
+	PaginationResponse
+}
+
+// ListPatientsMatchingCustomField - Get list of patients - matching custom-field criteria.
+// GET /v1/{practiceid}/patients/customfields/{customfieldid}/{customfieldvalue}
+// https://docs.athenahealth.com/api/api-ref/patient#Get-list-of-patients---matching-custom-field-criteria
+func (h *HTTPClient) ListPatientsMatchingCustomField(ctx context.Context, customFieldID, customFieldValue string) (*ListPatientsMatchingCustomFieldResult, error) {
+	out := &listPatientsMatchingCustomFieldResponse{}
+
+	_, err := h.Get(ctx, fmt.Sprintf("/patients/customfields/%s/%s", customFieldID, customFieldValue), nil, out)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ListPatientsMatchingCustomFieldResult{
+		Patients:   out.Patients,
+		Pagination: makePaginationResult(out.Next, out.Previous, out.TotalCount),
```

Comment:
This method isn't accepting pagination options so we will never be able to paginate this response.